### ISSUE TITLE
Processing of tables with identical columns.

### ISSUE THREAD

**Issue by RyanFleck** [Xalgorithms/lib-rules-int-scala#11]
When two tables have an identical column, i.e. **ID**, and both tables are combined, the **ID** column from the second table is discarded.

I am certain that some gov't tables will have at least one shared column, and that data cannot be discarded.

```yaml
ASSEMBLE overtime_table_A
  COLUMNS FROM table:hours
  COLUMNS FROM table:names;
```
[Example rule here, see *Section 2*.](https://github.com/Xalgorithms/lib-rules-int-scala/blob/rcf.testing/test-runs/overtime/match_names.rule)

One solution I can imagine is combining/filtering matching keys in the column. Another is to import and cross-product under a name with a number appended (but I'm not sure how well that will work with *writing* rules.)

**Comment by karfai** [comment]
In this example, the column with the same name **is not** used as a key for joining the tables and the data in the column is **different** in each table?

**Comment by RyanFleck** [comment]
No, the data in the column is identical, but stored in a different order as a test.

**Comment by karfai** [comment]
Ok... I'm not understanding what you're trying to do.... You want to **retain** both columns **without** using them as joining keys? If that's the case:

**Comment by RyanFleck** [comment]
Slack?

**Comment by karfai** [comment]
In fact, table joining assemblies don't work well at all. Both of the steps below should produce the same table, but one produces nothing and the other a cross-product without filtering:

```
# No join at all
ASSEMBLE full_table
  COLUMNS FROM table:a
  COLUMNS FROM table:b WHEN shared == @shared;

# empty table
ASSEMBLE half_table
  COLUMNS FROM table:a
  COLUMN roman FROM table:b WHEN shared == @shared;
 ```
